e2e/suite_test.go

```diff
@@ -317,16 +327,123 @@ var _ = Describe("pvc-autoresizer", func() {
+		}
+		resources = createPodPVC(resources, pvcName, sc, mode, pvcName, request, limit,
+			threshold, "", increase, storageLimit, initialResizeGroupByAnnotation, groupYLabel)
+
+		// Newly created small PVC
+		pvcName = "resize-group-pvc5"
+		request = "1Gi"
+		limit = "10Gi"
+		resources = createPodPVC(resources, pvcName, sc, mode, pvcName, request, limit,
+			threshold, "", increase, storageLimit, initialResizeGroupByAnnotation, groupXLabel)
+
+		By("checking the PVC size is mutated")
+		checkDiskResize(pvcName, "3Gi", true)
+
+	})
+
+	It("should not mutate the PVC size when the condition is not met", func() {
+		// large size PVC
+		pvcName := "non-resize-group-pvc1"
+		sc := "topolvm-provisioner-annotated"
+		mode := string(corev1.PersistentVolumeFilesystem)
+		request := "3Gi"
+		limit := "10Gi"
+		threshold := "50%"
+		increase := "1Gi"
+		storageLimit := ""
+		initialResizeGroupByAnnotation := "test-group"
+		groupXLabel := map[string]string{
+			initialResizeGroupByAnnotation: "group-x",
+		}
+		resources = createPodPVC(resources, pvcName, sc, mode, pvcName, request, limit,
+			threshold, "", increase, storageLimit, initialResizeGroupByAnnotation, groupXLabel)
+
+		// Newly created small PVC but different label key
+		pvcName = "non-resize-group-pvc2"
+		request = "1Gi"
+		limit = "10Gi"
+		initialResizeGroupByAnnotation = "test-group2"
+		groupXLabel2 := map[string]string{
+			initialResizeGroupByAnnotation: "group-x",
+		}
+		resources = createPodPVC(resources, pvcName, sc, mode, pvcName, request, limit,
+			threshold, "", increase, storageLimit, initialResizeGroupByAnnotation, groupXLabel2)
+
+		By("checking the PVC size is not mutated")
+		checkDoesNotResize(pvcName, "1Gi")
+
+		// The annotation is not set
+		pvcName = "empty-string-resize-group-pvc1"
+		request = "2Gi"
+		limit = "10Gi"
+		initialResizeGroupByAnnotation = ""
+		resources = createPodPVC(resources, pvcName, sc, mode, pvcName, request, limit,
+			threshold, "", increase, storageLimit, initialResizeGroupByAnnotation, groupXLabel)
+
+		pvcName = "empty-string-resize-group-pvc2"
+		request = "1Gi"
+		limit = "10Gi"
+		resources = createPodPVC(resources, pvcName, sc, mode, pvcName, request, limit,
+			threshold, "", increase, storageLimit, initialResizeGroupByAnnotation, groupXLabel)
+
+		By("checking the PVC size is not mutated")
+		checkDoesNotResize(pvcName, "1Gi")
+	})
 })
 
-func buildPodPVCTemplateYAML(pvcName, storageClassName, volumeMode, podName, request, limit, threshold, inodesThreshold, increase, storageLimit string) ([]byte, error) {
+func buildPodPVCTemplateYAML(ns, pvcName, storageClassName, volumeMode, podName, request, limit, threshold,
+	inodesThreshold, increase, storageLimit, initialResizeGroupByAnnotation string, labels map[string]string) ([]byte, error) {
 	var b bytes.Buffer
 	var err error
 
 	useAnnotation := "false"
 	if threshold != "" || increase != "" || storageLimit != "" {
 		useAnnotation = "true"
 	}
+	useLabel := "false"
```

Comment:
Why not use bool value directly?

Reply:
I fixed it.
(I was imitating other code in this function without deep thoughts...)